### PR TITLE
[SPARK-9369][SQL] Support IntervalType in UnsafeRow

### PR DESCRIPTION
**Pull request by cloud-fan**

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @rxin

File: sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java [hunk 1]
```diff
@@ -298,11 +299,6 @@ public UTF8String getUTF8String(int ordinal) {
   }
 
   @Override
-  public String getString(int ordinal) {
```
Comment (cloud-fan) [inline sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java]: it's the same with it's default implementation in `InternalRow`, so I think we don't need to reimplement it.

**Comment by rxin** [comment]
Can you update _readableFieldTypes in UnsafeRow?

**Comment by rxin** [comment]
LGTM other than that.

**Comment by cloud-fan** [comment]
looks like an un-related failure

**Comment by cloud-fan** [comment]
retest this please.

**Comment by cloud-fan** [comment]
retest this please.

**Comment by rxin** [comment]
Looks like there is a legitimate failure.

```
[info] - case between string and interval *** FAILED *** (26 milliseconds)
[info]   Code generation of CAST(interval -3 month 7 hours, IntervalType) failed:
[info]   java.util.concurrent.ExecutionException: java.lang.Exception: failed to compile:
```

**Comment by rxin** [comment]
Thanks - merging this.

**Pull request closed without merging**